nifi-nar-bundles/nifi-mqtt-bundle/nifi-mqtt-processors/src/main/java/org/apache/nifi/processors/mqtt/common/AbstractMQTTProcessor.java

```diff
@@ -0,0 +1,354 @@
+            .required(false)
+            .defaultValue("60")
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .build();
+
+    public static List<PropertyDescriptor> getAbstractPropertyDescriptors(){
+        final List<PropertyDescriptor> descriptors = new ArrayList<PropertyDescriptor>();
+        descriptors.add(PROP_BROKER_URI);
+        descriptors.add(PROP_CLIENTID);
+        descriptors.add(PROP_USERNAME);
+        descriptors.add(PROP_PASSWORD);
+        descriptors.add(PROP_SSL_CONTEXT_SERVICE);
+        descriptors.add(PROP_LAST_WILL_TOPIC);
+        descriptors.add(PROP_LAST_WILL_MESSAGE);
+        descriptors.add(PROP_LAST_WILL_RETAIN);
+        descriptors.add(PROP_LAST_WILL_QOS);
+        descriptors.add(PROP_CLEAN_SESSION);
+        descriptors.add(PROP_MQTT_VERSION);
+        descriptors.add(PROP_CONN_TIMEOUT);
+        descriptors.add(PROP_KEEP_ALIVE_INTERVAL);
+        return descriptors;
+    }
+
+    @Override
+    public Collection<ValidationResult> customValidate(final ValidationContext validationContext) {
+        final List<ValidationResult> results = new ArrayList<>(1);
+        final boolean usernameSet = validationContext.getProperty(PROP_USERNAME).isSet();
+        final boolean passwordSet = validationContext.getProperty(PROP_PASSWORD).isSet();
+
+        if ((usernameSet && !passwordSet) || (!usernameSet && passwordSet)) {
+            results.add(new ValidationResult.Builder().subject("Username and Password").valid(false).explanation("if username or password is set, both must be set").build());
+        }
+
+        final boolean lastWillTopicSet = validationContext.getProperty(PROP_LAST_WILL_TOPIC).isSet();
+        final boolean lastWillMessageSet = validationContext.getProperty(PROP_LAST_WILL_MESSAGE).isSet();
+
+        final boolean lastWillRetainSet = validationContext.getProperty(PROP_LAST_WILL_RETAIN).isSet();
+        final boolean lastWillQosSet = validationContext.getProperty(PROP_LAST_WILL_QOS).isSet();
+
+        // If any of the Last Will Properties are set
+        if(lastWillTopicSet || lastWillMessageSet || lastWillRetainSet || lastWillQosSet){
+            // And any are not set
+            if(!(lastWillTopicSet && lastWillMessageSet && lastWillRetainSet && lastWillQosSet)){
+                // Then mark as invalid
+                results.add(new ValidationResult.Builder().subject("Last Will Properties").valid(false).explanation("if any of the Last Will Properties (message, topic, retain and QoS) are " +
+                        "set, all must be set.").build());
+            }
+        }
+
+        return results;
+    }
+
+    public static Properties transformSSLContextService(SSLContextService sslContextService){
+        Properties properties = new Properties();
+        properties.setProperty("com.ibm.ssl.protocol", sslContextService.getSslAlgorithm());
```

Comment:
For consistency sake you may want to look how it is done in new Kafka https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-pubsub-processors/src/main/java/org/apache/nifi/processors/kafka/pubsub/PublishKafka.java#L100